lightly/models/utils.py

```diff
@@ -488,6 +487,10 @@ def cosine_decay_schedule(
         New momentum value to be used with update_momentum.
 
     """
+    assert step >= 0, "Current number can't be negative"
+    assert max_steps > 0, "Total epochs must be > 0"
+    assert step <= max_steps, f"Current step is bigger than total number of steps. Found {step} > {max_steps}."
```

Comment:
Are you sure that it is `step <= max_steps` and not `step < max_steps`? Because further down we have: `step / (max_steps - 1)`

Reply:
Yes, as I cut out the max_steps = 1 case with the if clause (and "step" starts from 0) and if I have step = max_steps = 1  is a case that I would like to keep

Reply:
Yes but the calculation doesn't work if we have for example, step=max_steps=4. Then the formula does: `step / (max_steps - 1) = 4 / 3` which is `>1` and we want the schedule to only go to 1 no?

Reply:
Oh yes sorry, step will never be = max step because as I wrote "step" starts from 0. Sorry